src/Logic.hs

```diff
@@ -369,16 +369,40 @@ handlePullRequestEdited prId newTitle newBaseBranch state =
     -- Do nothing if the pull request is not present.
     Nothing -> pure state
 
+-- Internal result type for parsing a merge command, which allows the
+-- consumer of `parseMergeCommand` to inspect the reason why a message
+-- was considered invalid.
+data ParseResult a
+  -- The parser found a valid prefix and a valid command.
+  = Success a
+  -- The parser found a valid prefix, but no valid command.
+  | Unknown Text
+  -- The parser decided to ignore the message because it did
+  -- not contain a valid prefix.
+  | Ignored
```

Comment:
If you use Haddock comments, these comments will show up when someone hovers over the constructors. There are multiple styles, but you usually want `--|` or `--^`

```haskell
-- | The data type X
data X 
-- | The constructor A
= A
| B 
-- ^ The constructor B
```